3/scripts.py

```diff
@@ -247,77 +247,51 @@ def steepest_2e(matrix, starting_sol):
     return best_sol, calculate_cost(best_sol, matrix)
 
 
-def get_random_neighbour_2n(solution, matrix):
-    neighbor = solution.copy()
-    all_nodes = set(range(matrix.shape[0]))
-    available_nodes = list(all_nodes - set(solution))
 
-    action = np.random.choice(['swap', 'replace'])
-
-    if action == 'swap':
-        i, j = np.random.choice(len(solution), size=2, replace=False)
-        neighbor[i], neighbor[j] = neighbor[j], neighbor[i]
-        delta = calculate_delta(solution, matrix, neighbor[i], j) + calculate_delta(solution, matrix, neighbor[j], i)
-    else:
-        i = np.random.choice(len(solution))
-        j = np.random.choice(len(available_nodes))
-        neighbor[i] = available_nodes[j]
-        delta = calculate_delta(solution, matrix, available_nodes[j], i)
-
-    return neighbor, delta
-
-
-def get_random_neighbour_2e(solution, matrix):
-    neighbor = solution.copy()
-    all_nodes = set(range(matrix.shape[0]))
-    available_nodes = list(all_nodes - set(solution))
-
-    action = np.random.choice(['swap', 'replace'])
-
-    if action == 'swap':
-        i, j = np.random.choice(len(solution), size=2, replace=False)
-
-        neighbor = np.concatenate((solution[:i], solution[i:j+1][::-1], solution[j+1:]))
-        delta = calculate_delta_edge(solution, matrix, i, j)
-
-    else:
-        i = np.random.choice(len(solution))
-        j = np.random.choice(len(available_nodes))
-        neighbor[i] = available_nodes[j]
-        delta = calculate_delta(solution, matrix, available_nodes[j], i)
+def greedy_2n(matrix, starting_sol):
+    best_sol = starting_sol
+    best_delta = 0
+    neighbourhood = get_neighbourhood_2n(starting_sol, matrix)
+    np.random.shuffle(neighbourhood)
+    i = 0
+    while i < len(neighbourhood):
+        probably_best_sol, best_delta = neighbourhood[i]
+        if best_delta >= 0:
+            i += 1
+        else:
+            best_sol, best_delta = neighbourhood[i]

```

Comment:
```suggestion
            best_sol = probably_best_sol
```

nie trzeba dwa razy czytać z neighborhooda